File: function-aws-alexa/features/test-aws-alexa-junit/skeleton/src/test/java/@defaultPackage.path@/ApplicationTest.java
package @defaultPackage@;

import com.amazon.ask.dispatcher.request.handler.HandlerInput;
import com.amazon.ask.model.*;
import com.amazon.ask.model.ui.OutputSpeech;
import io.micronaut.test.annotation.MicronautTest;
import org.junit.jupiter.api.Test;
import javax.inject.Inject;

import static org.junit.jupiter.api.Assertions.*;
import java.util.Optional;

@MicronautTest
public class ApplicationTest {

    @Inject
    Application handler;


    @Test
    public void testExampleHandler() throws Exception {
        final HandlerInput.Builder builder = HandlerInput.builder();
        final RequestEnvelope.Builder envelopeBuilder = RequestEnvelope.builder();
        envelopeBuilder.withRequest(IntentRequest.builder().build());
        builder.withRequestEnvelope(envelopeBuilder.build());
        final Optional<Response> response = handler.exampleHandler(builder.build());
        assertTrue(response.isPresent());
        final OutputSpeech outputSpeech = response.get().getOutputSpeech();
        assertTrue(
                outputSpeech.toString().contains(
                        "Example Response"
                )
        );
    }
}

File: function-aws-alexa/features/test-aws-alexa-spek/skeleton/src/test/kotlin/@defaultPackage.path@/ApplicationTest.kt
package @defaultPackage@

import io.micronaut.context.ApplicationContext
import io.micronaut.runtime.server.EmbeddedServer
import org.jetbrains.spek.api.Spek
import org.jetbrains.spek.api.dsl.describe
import org.jetbrains.spek.api.dsl.it
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertTrue
import com.amazon.ask.dispatcher.request.handler.HandlerInput
import com.amazon.ask.model.*
import com.amazon.ask.model.ui.OutputSpeech


class ApplicationTest: Spek({

    describe("test example handler") {
        val context = ApplicationContext.run()
        val handler = context.getBean(Application::class.java)
        val builder = HandlerInput.builder()
        val envelopeBuilder = RequestEnvelope.builder()
        envelopeBuilder.withRequest(IntentRequest.builder().build())
        builder.withRequestEnvelope(envelopeBuilder.build())
        val response = handler.exampleHandler(builder.build())

        it("response should be present") {
            assertTrue(response.isPresent())
        }

        val outputSpeech = response.get().getOutputSpeech()

        it ("contains the right speech") {
            assertTrue(
                outputSpeech.toString().contains(
                        "Example Response"
                )
            )
        }
        afterGroup {
            context.close()
        }
    }
})
